gnovm/stdlibs/std/emit_event.go

```diff
@@ -19,13 +19,26 @@ func X_emit(m *gno.Machine, typ string, attrs []string) {
 
 	_, pkgPath := currentRealm(m)
 	fnIdent := getPrevFunctionNameFromTarget(m, "Emit")
+	prevFunc := getPrevFunctionNameFromTarget(m, fnIdent) // get the previous function name of the function that called Emit
+
+	prevAddr, prevPath := X_getRealm(m, 1)
+	prev := prevStack{
+		PrevPkgPath: prevPath,
+		PrevPkgAddr: prevAddr,
+	}
+
+	ctx := m.Context.(ExecContext)
 
 	evt := gnoEvent{
-		Type:       typ,
+		OrigCaller: string(ctx.OrigCaller),
```

Comment:
You should call `ctx.OrigCaller.String()` here

Reply:
bf375c6acb9aa9e135a0f5388edd9e2a96f701a7